### ISSUE TITLE
Help with Illumina short reads

### ISSUE THREAD

**Issue by KristinaGagalova** [dfguan/purge_dups#18]
Hi, 
I have an assembly with short reads that we expect it's overassembled based on duplicated BUSCOs and estimated genome size.
I have an alignment of the short reads and I run ngscstat but I am having all zeros:

```
0       2435038856
1       0
2       0
3       0
4       0
5       0
6       0
7       0
8       0
9       0
10      0
11      0
12      0
13      0
14      0
15      0
16      0
...
```
Do you have any suggestions on what went wrong? Is maybe the insert size or some other parameter?
The bam file looks okay

Thank you in advance

**Comment by dfguan** [comment]
Hi, could share first 100K reads alignment with me?  So I can do a quick check on ngcstat program. Thanks, Dengfeng.

**Comment by KristinaGagalova** [comment]
Sure, the file is attached
[wpw1KReads.sam.gz](https://github.com/dfguan/purge_dups/files/3937789/wpw1KReads.sam.gz)

Thank you for checking that!

**Comment by dfguan** [comment]
Hi, ngcstat is now only taking unsorted bam file as input for paired-end reads, I think your bam is sorted? Could you please try it with the unsorted bam?  Dengfeng.

**Comment by KristinaGagalova** [comment]
The bam file is sorted. I will try with completely unsorted bam and let you know.
Why are you taking an unsorted bam, are you randomly sampling the coverage from all the scaffolds/contigs?

**Comment by dfguan** [comment]
I only use unambiguously mapped reads for coverage calculation,  I don't know if there is a flag for that, so I need reads organized by their names. Make sense? Dengfeng.

**Comment by KristinaGagalova** [comment]
The output for the unsorted reads is the same
```
0       2435038856
1       0
2       0
3       0
4       0
5       0
6       0
7       0
8       0
9       0
10      0
11      0
12      0
13      0
14      0
15      0
16      0
```
Do the reads need to be sorted by name instead? With this command
```
samtools sort -n input -o output
```

**Comment by dfguan** [comment]
How do you unsorted bam look like? Could you show me 5 reads? Thanks, Dengfeng.

**Comment by KristinaGagalova** [comment]
Hi, 
I have few test runs but none of those is outputting the reads coverage profile in TX.stat and TX.base.cov.

This is the unsorted sam
[wpwLR.unsorted.sam.gz](https://github.com/dfguan/purge_dups/files/3946170/wpwLR.unsorted.sam.gz)
and this one is the name sorted with samtools
[wpwLR.nameSor.sam.gz](https://github.com/dfguan/purge_dups/files/3946171/wpwLR.nameSor.sam.gz)
Consider that the unsorted is run with multithreading which shuffles the reads.

Is the contigs names parsing correct? I have 167,268 scaffolds for my alignment, do you have any limitation on that?
Thank you in advance for the help

**Comment by dfguan** [comment]
Hi, is this single-read dataset? I saw none reads are paired up.

**Comment by KristinaGagalova** [comment]
That's weird, I am not sure if it's because of the -p option in bwa mem. I will try to re-run it.
Is actually the tools working with paired-end reads only?

**Comment by dfguan** [comment]
Yes, it is only working with paired-end reads now, although it should support single-end reads. I don't have time to work on that now, but I will put that to my todo list.
